ngi_pipeline/engines/qc_ngi/launchers.py

```diff
@@ -0,0 +1,144 @@
+"""Run the qc pipeline on fastq samples; currently consists of FastQC and
+fastq_screen."""
+
+import os
+import subprocess
+
+from ngi_pipeline.engines.qc_ngi.workflows import return_cl_for_workflow
+from ngi_pipeline.log.loggers import log_process_non_blocking, minimal_logger
+from ngi_pipeline.utils.classes import with_ngi_config
+from ngi_pipeline.utils.filesystem import rotate_file, safe_makedir
+from ngi_pipeline.utils.parsers import find_fastq_read_pairs
+
+LOG = minimal_logger(__name__)
+
+
+@with_ngi_config
+def analyze(project, sample, config=None, config_file_path=None):
+    """The main entry point for the qc pipeline."""
+    LOG.info("Processing project/sample {}/{}".format(project, sample))
+
+    # Two paths diverged in a yellow wood
+    project_analysis_path = os.path.join(project.base_path,
+                                         "ANALYSIS",
+                                         project.project_id,
+                                         "qc_ngi")
+    # and sorry I could not travel both
+    sample_analysis_path = os.path.join(project_analysis_path, sample.name)
+    # and be one traveler, long I stood
+    log_dir_path = os.path.join(project_analysis_path, "logs")
+    # and looked down one as far as I could
+    safe_makedir(sample_analysis_path)
+    # To where it bent in the undergrowth
+    safe_makedir(log_dir_path)
+    # I need to go to sleep
```

Comment:
You Europeans never know our poets. But you're right, I was starting to lose it.